src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -243,11 +286,44 @@ private static CharSequence getPokedPattern() {
     }
 
     /**
+     * Get a list of possible permutations for a valid doPeek
```

Comment:
Below is my implementation of `getPermutations`. It uses a 0-1 knapsack algorithm purposed for creating combinations.